connector/src/main/java/org/geysermc/connector/utils/LoginEncryptionUtils.java

```diff
@@ -176,132 +179,118 @@ private static void sendEncryptionFailedMessage(GeyserConnector connector) {
-                .build();
+    public static void buildAndShowLoginDetailsWindow(GeyserSession session) {
+        session.sendForm(
+                CustomForm.builder()
+                        .translator(LanguageUtils::getPlayerLocaleString, session.getLocale())
+                        .title("geyser.auth.login.form.details.title")
+                        .label("geyser.auth.login.form.details.desc")
+                        .input("geyser.auth.login.form.details.email", "account@geysermc.org", "")
+                        .input("geyser.auth.login.form.details.pass", "123456", "")
+                        .responseHandler((form, responseData) -> {
+                            CustomFormResponse response = form.parseResponse(responseData);
+                            if (!response.isCorrect()) {
+                                buildAndShowLoginDetailsWindow(session);
+                                return;
+                            }
 
-        session.sendForm(window, AUTH_DETAILS_FORM_ID);
+                            session.authenticate(response.next(), response.next());
+                        }));
     }
 
     /**
-     * Prompts the user between either OAuth code login or manual password authentication
+     * Promts the user between either OAuth code login or manual password authentication
      */
-    public static void showMicrosoftAuthenticationWindow(GeyserSession session) {
-        String userLanguage = session.getLocale();
-        SimpleFormWindow window = new SimpleFormWindow(LanguageUtils.getPlayerLocaleString("geyser.auth.login.form.notice.btn_login.microsoft", userLanguage), "");
-        window.getButtons().add(new FormButton(LanguageUtils.getPlayerLocaleString("geyser.auth.login.method.browser", userLanguage)));
-        window.getButtons().add(new FormButton(LanguageUtils.getPlayerLocaleString("geyser.auth.login.method.password", userLanguage))); // This form won't show if password authentication is disabled
-        window.getButtons().add(new FormButton(LanguageUtils.getPlayerLocaleString("geyser.auth.login.form.notice.btn_disconnect", userLanguage)));
-        session.sendForm(window, AUTH_MSA_DETAILS_FORM_ID);
+    public static void buildAndShowMicrosoftAuthenticationWindow(GeyserSession session) {
+        session.sendForm(
+                SimpleForm.builder()
+                        .translator(LanguageUtils::getPlayerLocaleString, session.getLocale())
+                        .title("geyser.auth.login.form.notice.btn_login.microsoft")
+                        .button("geyser.auth.login.method.browser")
+                        .button("geyser.auth.login.method.password")
+                        .button("geyser.auth.login.form.notice.btn_disconnect")
+                        .responseHandler((form, responseData) -> {
+                            SimpleFormResponse response = form.parseResponse(responseData);
+                            if (!response.isCorrect()) {
```

Comment:
Not related to Floodgate, but if isCorrect could also mean that the client can exit out of the menu, I'm not sure if that's a great name forit.